pyglet/window/__init__.py

```diff
@@ -1017,7 +1017,10 @@ def set_size(self, width, height):
                 New height of the window, in pixels.
 
         """
-        raise NotImplementedError('abstract')
+        if self._fullscreen:
+            raise WindowException('Cannot set size of fullscreen window.')
+        self._width = max(width, 1)
+        self._height = max(height, 1)
```

Comment:
I'm not sure if this is worth doing here or if it should just be:
```python
self._width = width
self._height = height
```
The cocoa version did it with max(), but xlib just takes the user input. Win32 doesn't update these values on calls to set_size at all

Reply:
We definitely want to do the `max` call here to be safe. Some OpenGL drivers will error out with zero values for width and height (such as viewport settings). It's common for users to use the `on_resize` method to update these things, so we don't want to allow a zero value.

Reply:
Ok cool, I'll go ahead and leave that in then

Reply:
The other option is to raise an exception if the user entered a value less than 1. Which is the preferred method?

Reply:
That's a good point. Raising a `WindowException` (or similar) would also be acceptable.